r2-core/src/main/java/com/linkedin/r2/transport/http/client/AsyncPoolImpl.java

```diff
@@ -374,8 +374,12 @@ public Cancellable get(final Callback<T> callback)
           {
             if (_waiters.size() < _maxWaiters)
             {
+              if (isWaiterTimeoutWithinFastFailThreshold())
```

Comment:
This check should also exist within the constructor and offspring factory where it is passed in. We do address this in a separate PR if you'd like.

Reply:
Why this needs to be there in other places ? checking here should take care of all the cases right?

Reply:
We should not silently ignore configs passed in. If we could fail early in HttpClientFactory.Builder or the Offspring factory user can detect much earlier.

Reply:
Understood. I will create a separate PR for this.